modules/photo/src/merge.cpp

```diff
@@ -176,16 +176,16 @@ class MergeMertensImpl CV_FINAL : public MergeMertens
 
         parallel_for_(Range(0, static_cast<int>(images.size())), [&](const Range& range) {
             for(int i = range.start; i < range.end; i++) {
-                Mat& img = images[i];
-                Mat gray, contrast, saturation, wellexp;
+                Mat img, gray, contrast, saturation, wellexp;
                 std::vector<Mat> splitted(channels);
 
-                img.convertTo(img, CV_32F, 1.0f/255.0f);
+                images[i].convertTo(img, CV_32F, 1.0f/255.0f);
                 if(channels == 3) {
                     cvtColor(img, gray, COLOR_RGB2GRAY);
                 } else {
                     img.copyTo(gray);
                 }
+                images[i] = img;
```

Comment:
Nit: Moving this up to line 183 may make it a bit clearer, that we just want a shallow copy here.